src/sinks/file.rs

```diff
@@ -0,0 +1,316 @@
+use crate::{
+    buffers::Acker,
+    event::Event,
+    sinks::util::{
+        encoding::BasicEncoding,
+        file::{EmbeddedFileSink, FileSink},
+        SinkExt,
+    },
+    template::Template,
+    topology::config::DataType,
+};
+
+use futures::{future, Async, AsyncSink, Sink, StartSend};
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+use std::path::PathBuf;
+use std::time::Instant;
+
+#[derive(Deserialize, Serialize, Debug)]
+#[serde(deny_unknown_fields)]
+pub struct FileSinkConfig {
+    #[serde(rename = "path")]
+    pub path_template: String,
+    #[serde(rename = "timeout", default = "default_close_timeout_secs")]
```

Comment:
why rename this here? I also believe the default impl for Option is None

Reply:
`close_timeout_secs` seems to long to write for end-user, but I can rename it
good point about `None`